### ISSUE TITLE
Custom Frequency Input for Handheld Radios

### ISSUE THREAD

**Issue by lule345** [Pixeltheaertist/Trieste-Port-14#102]
## Description

As of right now, handheld transceivers/radios broadcast on one dedicated frequency (one for regular handheld (salvage/emergency) and one for security (not to be confused with ":s")). Radios, with this issue implemented, should be capable of tuning between frequencies (possibly AM band with 540 kHz to 1700 kHz?), as well as saving said channels in memory, much like how Barotrauma's radio (chat) system works. This would allow for private communications between crew members to occur, whether it'd be for clown/mime shenanigans or for traitor terrorism, as well as provide additional immersion for roleplay.
![image](https://github.com/user-attachments/assets/f1d5ed41-8f1b-4a80-8fa6-d4301f85d399)

**Comment by Pixeltheaertist** [comment]
@nolius i know at one point you offered something like this, idk if it's still possible

**Comment by Pixeltheaertist** [comment]
sorry, @noliuss

**Comment by Noliuss** [comment]
Yeah, you can use it if you want, needed commits are Noliuss/Barotrauma-Station@e5a087b188e907d559bf00c1270345049de2bdcc and Noliuss/Barotrauma-Station@bdb777dd136b860792d8db4ba77e348bb5f19bce if i recall correctly. Barotrauma Station is licensed under AGPL-3.0, but it is fine, if someone boring enough will be bothering you about that just show them this comment. I've made it so that each headset uses random frequency from the start because i don't want people from outpost and submarine to communicate from the start, but maybe you'd want to change it. Anyways, feel free to ask me if you need any help with porting/changing it

**Comment by Noliuss** [comment]
No, with those two all of them use the same frequency. It was added with this commit: Noliuss/Barotrauma-Station@0315f7993e9b86e7eae428efdf0f87c6f28acdc0. Also note that in first commit i've removed from Resources/Locale/en-US/headset/headset-component.ftl {$channel} so it looks more like in barotrauma and doesn't say which channel does someone use. You probably would want to change it back

**Comment by Pixeltheaertist** [comment]
i'll check it out, thanks!

**Comment by quri1q** [comment]
Im pretty sure frontier station also has a similar concept with their radios
